core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -361,6 +362,13 @@ Schema getOutputSchema(Map<String, Schema> inputSchemas, FailureCollector collec
 
     return outputFieldInfo.values();
   }
+  private boolean inputSchemasAvailable(Map<String, Schema> inputSchemas) {
+    // TODO: Remove isEmpty() check when CDAP-16351 is fixed
+    if (inputSchemas.isEmpty() || inputSchemas.values().stream().anyMatch(v -> v == null)) {
+      return false;
```

Comment:
can change this to a single line:

`return !inputSchemas.isEmpty() && ...`

Reply:
done.